docs/credentials_and_secrets.md

```diff
@@ -0,0 +1,72 @@
+# Credentials and Local Secrets
+
+Rails credentials allows you to securely store sensitive configuration information and credentials for your Rails application. This feature is particularly useful for storing things like API keys, database passwords, or any other confidential data that your application needs to access.
+
+Local secrets (environment variables) are secrets that are environment-specific information unique to your local system or the cloud platform you use such as Heroku. We use these variables when every local system requires different credentials. They can also be used to quickly change certain settings that might change very often like disabling a certain feature temporarily. There is no need to update any code and re-deploy the app in this case.
+
+## Credentials
+
+We use the default rails credentials file to store application credentials that can be accessed only by the app with the requirement of having a `master.key` file. This file will reside only on your local and your cloud platform as a config variable named `RAILS_MASTER_KEY`. The `master.key` file should not be pushed to github for security purposes.
```

Comment:
this should be a given here. but talk about how the setup will help you set the key up when onboarding. 

if they were to do this manually they would need to create the key file. 
* use the same value for the app if this is not a new project so they can decrypt the existing file
* otherwise how to generate a new credentials file replacing the current one for a new project. (give manual instructions for the command we run in the setup but how they can do that manually without running the setup script)